dbx/utils/common.py

```diff
@@ -176,17 +176,32 @@ def get_all_environment_names(self) -> Any:
 
 
 class Jinja2DeploymentConfig(AbstractDeploymentConfig):
-    def __init__(self, path, ext):
+    def __init__(self, path, ext, _template_variables_file=None):
         super().__init__(path)
         self._ext = ext
+        self._template_variables_file = _template_variables_file
 
     def _render_jinja_template(self) -> str:
         path_list = self._path.split("/")
         file_name = path_list.pop()
         file_path = "/".join(path_list)
 
         j2_env = jinja2.Environment(loader=jinja2.FileSystemLoader(file_path))
-        return j2_env.get_template(file_name).render(os.environ)
+
+        if self._template_variables_file:
```

Comment:
I think that the template rendering code could be moved outside the if-else, like:

```python
if self._template_variables_file:
    with open(self._template_variables_file, "r") as file:
        template_vars = ruamel.yaml.safe_load(file)
    template_vars = {**os.environ, **template_vars}
else:
    dbx_echo(
        "No custom template variables file provided."
        "Using only environment variables to render Jinja template."
    )
    template_vars = dict(os.environ)

rendered_template = j2_env.get_template(file_name).render(template_vars)
```